packages/components/inputs/time-input/src/time-input-body.js

```diff
@@ -75,20 +74,29 @@ export default class TimeInputBody extends React.Component {
             aria-readonly={this.props.isReadOnly}
             contentEditable={!this.props.isReadOnly}
           />
-          <ClearSection
-            isDisabled={this.props.isDisabled}
-            hasError={this.props.hasError}
-            isReadOnly={this.props.isReadOnly}
-            onClear={this.props.onClear}
-          />
+
+          {!this.props.isDisabled && !this.props.isReadOnly && (
```

Comment:
Thanks to adjustments in the styling, we don't need to display this component anymore when it's not supposed to be shown.

It was being kept because of the borders around it, but that was fixed and all the borders are now in the container.